storage/s3.go

```diff
@@ -44,23 +49,14 @@ const (
 	gcsEndpoint = "storage.googleapis.com"
 )
 
-// Re-used AWS sessions dramatically improve performance.
-var cachedS3 *S3
-
-// Init creates a new global S3 session.
-func Init(opts S3Options) error {
-	s3, err := NewS3Storage(opts)
-	cachedS3 = s3
-	return err
-}
-
 // S3 is a storage type which interacts with S3API, DownloaderAPI and
 // UploaderAPI.
 type S3 struct {
-	api         s3iface.S3API
-	downloader  s3manageriface.DownloaderAPI
-	uploader    s3manageriface.UploaderAPI
-	endpointURL urlpkg.URL
+	api           s3iface.S3API
+	downloader    s3manageriface.DownloaderAPI
+	uploader      s3manageriface.UploaderAPI
+	endpointURL   urlpkg.URL
+	destinationS3 *S3
```

Comment:
 Why do we need to embed destination S3? If the purpose here is to use destination session on [S3 Copy operation](https://github.com/peak/s5cmd/blob/7d53d22338a8315e28942bb5e44214e3cf013c41/storage/s3.go#L342), that means we can safely create destination client rather than source client on [doCopy()](https://github.com/peak/s5cmd/blob/master/command/cp.go#L446) 

Reply:
hmm, i thought if we don't keep source and destination clients attached we might need to pass destination options for each function call from upper level command down to Copy() call.